### PR TITLE
(SIMP-4009) Add Hiera 5 hiera.yaml

### PR DESCRIPTION
**Pull request by jeefberkey**

### CODE REVIEW

File: build/simp-environment.spec [hunk 1]
```diff
@@ -33,17 +33,14 @@
...
 Release: 0%{?dist}
 License: Apache License 2.0
 Group: Applications/System
 Source: %{name}-%{version}-%{release}.tar.gz
 Buildroot: %{_tmppath}/%{name}-%{version}-%{release}-buildroot
 Requires: libselinux-utils
 Requires: policycoreutils
-Requires: pupmod-simp-simp >= 3.0.0
-Requires: pupmod-simp-pki >= 6.0.0-0
-Requires: createrepo
```
Comment (lnemsick-simp) [inline build/simp-environment.spec]: createrepo is used in the %post.  Why did you remove this dependency?

Comment (trevor-vaughan) [reply build/simp-environment.spec]: To be more clear, this could be changed to `Requires(post): createrepo`

File: build/simp-environment.spec [hunk 2]
```diff
@@ -33,17 +33,14 @@
...
+Version: 6.3.0
 Release: 0%{?dist}
 License: Apache License 2.0
 Group: Applications/System
 Source: %{name}-%{version}-%{release}.tar.gz
 Buildroot: %{_tmppath}/%{name}-%{version}-%{release}-buildroot
 Requires: libselinux-utils
 Requires: policycoreutils
-Requires: pupmod-simp-simp >= 3.0.0
-Requires: pupmod-simp-pki >= 6.0.0-0
```
Comment (trevor-vaughan) [inline build/simp-environment.spec]: We probably need to implement the optional dependency checking before ripping this out. The automatic RPM dependencies are only added if you're using the SIMP RPM build system.

https://github.com/simp/pupmod-simp-simplib/pull/148

**Comment by jeefberkey** [comment]
Check SIMP-2272 to rename `hieradata` to `data`

File: build/simp-environment.spec [hunk 3]
```diff
@@ -33,26 +33,24 @@
...
 Group: Applications/System
 Source: %{name}-%{version}-%{release}.tar.gz
 Buildroot: %{_tmppath}/%{name}-%{version}-%{release}-buildroot
 Requires: libselinux-utils
+Requires: openssl
 Requires: policycoreutils
-Requires: pupmod-simp-simp >= 3.0.0
-Requires: pupmod-simp-pki >= 6.0.0-0
-Requires: createrepo
+Requires: rubygem(simp-cli) >= 1.0.0-0
```
Comment (lnemsick-simp) [inline build/simp-environment.spec]: Did you mean to require simp-adapter?  I don't understand why simp-cli is a dependency.

**Comment by lnemsick-simp** [comment]
I will make updates for SIMP-2272 (which allows staged upgrades to environment hiera.yaml use) and to remove OBE compliance_profiles

File: build/simp-environment.spec [hunk 4]
```diff
@@ -42,11 +42,11 @@ Buildroot: %{_tmppath}/%{name}-%{version}-%{release}-buildroot
 Requires: libselinux-utils
 Requires: openssl
 Requires: policycoreutils
-Requires: simp-rsync >= 6.2.0-0
-Requires: simp-utils >= 6.0.0-0
 Requires(pre,preun,post,postun): simp-adapter
+Requires(pre,post): puppet
 Requires(post): coreutils
 Requires(post): createrepo
+Requires(post): facter
```
Comment (jeefberkey) [inline build/simp-environment.spec]: I don't believe there is still a facter rpm

Comment (lnemsick-simp) [reply build/simp-environment.spec]: that doesn't mean facter must be in a rpm (same with puppet).  It means the puppet-agent rpm advertises that it provides facter and puppet.  (rpm -q --provides puppet-agent to see what it provides)

Comment (jeefberkey) [reply build/simp-environment.spec]: Makes sense.

**Comment by lnemsick-simp** [comment]
I'll fix
